Python/optimizer_symbols.c

```diff
@@ -164,12 +164,15 @@ _Py_uop_sym_set_const(_Py_UopsSymbol *sym, PyObject *const_val)
     return true;
 }
 
-
 bool
 _Py_uop_sym_set_null(_Py_UopsSymbol *sym)
 {
+    if (_Py_uop_sym_is_not_null(sym)) {
+        sym_set_bottom(sym);
+        return false;
+    }
     sym_set_flag(sym, IS_NULL);
-    return !_Py_uop_sym_is_bottom(sym);
+    return true;
```

Comment:
It is not a refactoring.
Calling `_Py_uop_sym_set_null` on a non-NULL symbol would fail an assertion in `_Py_uop_sym_is_bottom`

And yes, it should be applied to `set_non_null` as well.